app/controllers/concerns/wizard_steps.rb

```diff
@@ -0,0 +1,66 @@
+module WizardSteps
+  extend ActiveSupport::Concern
+
+  included do
+    class_attribute :wizard_class
+    before_action :load_wizard, :load_current_step, except: %i[index completed]
```

Comment:
Sorry, don't quite get this...

Reply:
Sure, I'm not sure I described it well. The essence is that I would remove the `before_action` filters from the concern. Because they could have an unpredictable behaviour when included in a controller and they are really not needed. 

Of course if you remove them you need to load the wizard and current step somehow, which is what the 2 methods above would be for. To lazy load and memoize the relevant records.